main/src/ca/uwaterloo/flix/api/lsp/provider/SemanticTokensProvider.scala

```diff
@@ -0,0 +1,99 @@
+package ca.uwaterloo.flix.api.lsp.provider
+
+import ca.uwaterloo.flix.api.lsp.{Entity, Index}
+import ca.uwaterloo.flix.language.ast.SourceLocation
+import ca.uwaterloo.flix.language.ast.TypedAst.{Expression, Root}
+import org.json4s.JsonAST.JObject
+import org.json4s.JsonDSL._
+
+import scala.collection.mutable.ArrayBuffer
+
+object SemanticTokensProvider {
+  /**
+   * The token types currently supported by the provider.
+   *
+   * Note: The intermediate TypeScript server is responsible for communicating the legend
+   * to the client. The TS server must list the token types in the same order as is used
+   * here in the Scala code. The reason for this is that the response is encoded as a
+   * sequence of integers, and the integer representing a token type is the type's
+   * index in the legend.
+   */
+  private object TokenType extends Enumeration {
```

Comment:
I would represent this in a separate file like other enums, e.g.: https://github.com/flix/flix/blob/master/main/src/ca/uwaterloo/flix/api/lsp/DocumentHighlightKind.scala